yapapi/executor/__init__.py

```diff
@@ -185,14 +185,36 @@ async def run_worker(work_context: WorkContext) -> None:
             with work_queue.new_consumer() as consumer:
                 try:
 
-                    async def task_generator() -> AsyncIterator[Task[D, R]]:
+                    # the `task_generator` here is what feeds the user's `worker` directly
```

Comment:
```suggestion
                    # the `task_generator` here is passed as an argument to the user's `worker` function
```
(I hope this is true? If not, then pls change the comment some other way so I will understand what does it mean something "feeds" the `worker`).

Reply:
will confirm to be 100% sure but yes, I'm 99% sure you're correct here